webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js

```diff
@@ -81,15 +88,31 @@ const CVErrataDateFilterContent = ({
   const resetFilters = () => {
     setStartDate(convertAPIDateToUIFormat(ruleStartDate));
     setEndDate(convertAPIDateToUIFormat(ruleEndDate));
-    setSelectedTypes(types);
     setDateType(ruleDateType);
+    setSelectedTypes(getInitialSelectedTypes());
+  };
+
+  const getInitialSelectedTypes = () => {
+    if (allow_other_types) {
+      return[...types, 'other'];
+    } else {
+      return types;
+    }
   };
 
   const onTypeSelect = (selection) => {
+    let updatedSelectedTypes;
     if (selectedTypes.includes(selection)) {
+      // If the selection is the only selection remaining, do not allow it to be removed
       if (selectedTypes.length === 1) return;
-      setSelectedTypes(selectedTypes.filter(e => e !== selection));
-    } else setSelectedTypes([...selectedTypes, selection]);
+
+      // Filter out the current selection from the selected types to update
+      updatedSelectedTypes = selectedTypes.filter(e => e !== selection);
+    } else {
+      updatedSelectedTypes = [...selectedTypes, selection];
+    }
+
+    setSelectedTypes(updatedSelectedTypes);
```

Comment:
Don't forget to (un)do the refactor here.

Reply:
Done! I left some comments for future readability but it should otherwise be unchanged.